spec/services/csat_survey_service_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+describe CsatSurveyService do
+  let(:account) { create(:account) }
+  let(:inbox) { create(:inbox, account: account, csat_survey_enabled: true) }
+  let(:conversation) { create(:conversation, inbox: inbox, account: account, status: :resolved) }
+  let(:service) { described_class.new(conversation: conversation) }
+
+  describe '#perform' do
+    let(:csat_template) { instance_double(MessageTemplates::Template::CsatSurvey) }
+
+    before do
+      allow(MessageTemplates::Template::CsatSurvey).to receive(:new).and_return(csat_template)
+      allow(csat_template).to receive(:perform)
+      allow(conversation).to receive(:create_csat_not_sent_activity_message)
+    end
+
+    context 'when CSAT survey should be sent' do
+      before do
+        allow(conversation).to receive(:can_reply?).and_return(true)
+      end
+
+      it 'sends CSAT survey when within messaging window' do
+        service.perform
+
+        expect(MessageTemplates::Template::CsatSurvey).to have_received(:new).with(conversation: conversation)
```

Comment:
```suggestion
        expect(MessageTemplates::Template::CsatSurvey).to have_received(:new).with(conversation: conversation)
        expect(csat_template).to have_received(:perform)
```

Let's also ensure that perform is called.